index.js

```diff
@@ -200,24 +263,121 @@ function numEventLoopLags() {
       minBucketLowerBound = Math.min(minBucketLowerBound, snapshot.ranges[i]);
     }
   }
-  return {numHangs: numHangs, minBucketLowerBound: minBucketLowerBound};
+  panel.port.emit("warning", null);
+  return Promise.resolve({numHangs: numHangs, minBucketLowerBound: minBucketLowerBound});
 }
 
 function numInputEventResponseLags() {
   try {
     var snapshot = Services.telemetry.getHistogramById("INPUT_EVENT_RESPONSE_MS").snapshot();
   } catch (e) { // histogram doesn't exist, the Firefox version is likely older than 46.0a1
-    return {numHangs: null, minBucketLowerBound: 0};
+    panel.port.emit("warning", "unavailableInputEventResponseLags");
+    return Promise.resolve({numHangs: null, minBucketLowerBound: 0});
   }
   let numHangs = 0;
   let minBucketLowerBound = Infinity;
   for (let i = 0; i < snapshot.ranges.length; ++i) {
     if (snapshot.ranges[i] > gHangThreshold) {
-      result += snapshot.counts[i];
+      numHangs += snapshot.counts[i];
       minBucketLowerBound = Math.min(minBucketLowerBound, snapshot.ranges[i]);
     }
   }
-  return {numHangs: numHangs, minBucketLowerBound: minBucketLowerBound};
+  return Promise.resolve({numHangs: numHangs, minBucketLowerBound: minBucketLowerBound});
+}
+
+// Returns an array of the most recent BHR hangs, or null on error
```

Comment:
Not valid anymore. It returns a promise.

For the record, correct practice would have us reject the promise on error, not resolve to an exceptional value like null or -1.